src/basic/parse-util.c

```diff
@@ -779,3 +780,12 @@ int parse_loadavg_fixed_point(const char *s, loadavg_t *ret) {
 
         return store_loadavg_fixed_point(i, f, ret);
 }
+
+int parse_slot(const char *s, int *ret, int nonzero) {
+        assert(s);
+
+        if (nonzero && s[0] == '0')
+                return -EINVAL;
+
+        return safe_atoi(s, ret);
```

Comment:
if you add a helper like this, you might as well refuse <= 0 slots here, instead of doing this in the caller where it is currently done

Reply:
Will do.